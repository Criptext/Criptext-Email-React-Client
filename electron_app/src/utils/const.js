/* process */
require('dotenv').config();

const API_CLIENT_VERSION = '9.0.0';
const LINK_DEVICES_FILE_VERSION = '3';
const PROD_SOCKET_URL = 'wss://socket.criptext.com';
const PROD_SERVER_URL = 'https://api.criptext.com';
const PROD_DATA_TRANSFER_URL = 'https://transfer.criptext.com';
const FILE_SERVER_APP_ID = 'qynhtyzjrshazxqarkpy';
const FILE_SERVER_KEY = 'lofjksedbxuucdjjpnby';
const PROD_APP_DOMAIN = 'criptext.com';
const NEWS_SERVER_URL = 'https://news.criptext.com';

const isDevelopment = process.env.NODE_ENV === 'development';
if (isDevelopment) {
  console.log('\x1b[33m%s\x1b[0m', `\n\t Mode: ${process.env.NODE_ENV}\n`);
}

module.exports = {
  FILE_SERVER_APP_ID,
  FILE_SERVER_KEY,
  APP_DOMAIN: isDevelopment ? process.env.DEV_APP_DOMAIN : PROD_APP_DOMAIN,
  SERVER_URL: isDevelopment ? process.env.DEV_SERVER_URL : PROD_SERVER_URL,
  SOCKET_URL: isDevelopment ? process.env.DEV_SOCKET_URL : PROD_SOCKET_URL,
  DATA_TRANSFER_URL: isDevelopment
    ? process.env.DEV_DATA_TRANSFER_URL
    : PROD_DATA_TRANSFER_URL,
  NEWS_SERVER_URL,
  API_CLIENT_VERSION,
  LINK_DEVICES_FILE_VERSION
};
