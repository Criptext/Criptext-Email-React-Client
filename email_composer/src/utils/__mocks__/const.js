export const appDomain = 'criptext.com';

export const composerEvents = {
  EDIT_DRAFT: 'edit-draft',
  FORWARD: 'forward',
  REPLY: 'reply',
  REPLY_ALL: 'reply-all',
  NEW_WITH_DATA: 'new-with-data'
};

export const externalDomains = ['gmail.com', 'hotmail.com', 'yahoo.com'];
