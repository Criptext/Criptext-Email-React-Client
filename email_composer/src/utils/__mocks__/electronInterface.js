export const myAccount = {
  id: 1,
  recipientId: 'myAccountRecipientId',
  name: 'myAccountName'
};

export const emailKey = '1';
export const emailKeyWithFile = '2';

export const composerEvents = {
  EDIT_DRAFT: 'edit-draft',
  REPLY: 'reply',
  REPLY_ALL: 'reply-all',
  FORWARD: 'forward'
};

export const mySettings = {
  language: 'en',
  opened: 1,
  theme: 'light'
};
