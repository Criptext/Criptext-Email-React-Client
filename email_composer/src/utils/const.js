/*global process */
export const appDomain =
  process.env.NODE_ENV === 'development'
    ? process.env.REACT_APP_APPDOMAIN
    : 'criptext.com';

export const avatarBaseUrl =
  process.env.NODE_ENV === 'development'
    ? process.env.REACT_APP_AVATAR_URL
    : `https://api.criptext.com/user/avatar/`;

export const composerEvents = {
  EDIT_DRAFT: 'edit-draft',
  FORWARD: 'forward',
  REPLY: 'reply',
  REPLY_ALL: 'reply-all',
  NEW_WITH_DATA: 'new-with-data'
};

export const defaultEmptyMimetypeValue = 'application/octet-stream';

export const previewLength = 100;
