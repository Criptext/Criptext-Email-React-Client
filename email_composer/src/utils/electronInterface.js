import { labels } from './systemLabels';
const electron = window.require('electron');
const { remote, ipcRenderer } = electron;
const composerId = remote.getCurrentWindow().id;
const globalManager = remote.require('./src/globalManager');
export const getAlicePort = remote.require('./src/aliceManager').getPort;
export const getAlicePassword = remote.require('./src/aliceManager')
  .getPassword;

export const { FILE_SERVER_APP_ID, FILE_SERVER_KEY } = remote.require(
  './src/utils/const'
);

export const getEmailToEdit = () => {
  return globalManager.emailToEdit.get(composerId);
};

const accounts = remote.require('./src/Account');
export let myAccount = accounts.activeAccount;
export const setMyAccount = recipientId => {
  myAccount = accounts.find(account => account.recipientId === recipientId);
};
export const mySettings = remote.require('./src/Settings');
export const LabelType = labels;

/* Window events
   ----------------------------- */
export const sendEventToMailbox = (name, params) => {
  ipcRenderer.send(name, params);
};
