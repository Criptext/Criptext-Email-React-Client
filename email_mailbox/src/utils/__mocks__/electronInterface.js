export const openEmptyComposerWindow = () => {};

export const myAccount = {
  id: 1,
  recipientId: 'test',
  isActive: true,
  name: 'User test',
  logged: { 2: { id: 2, name: 'User 2', recipientId: 'user2', isActive: 0 } }
};

export const mySettings = { language: 'en' };

export const LabelType = {
  inbox: {
    id: 1,
    text: 'Inbox',
    color: '0091ff'
  },
  spam: {
    id: 2,
    text: 'Spam',
    color: 'ff0000'
  },
  sent: {
    id: 3,
    text: 'Sent',
    color: '1a9759'
  },
  starred: {
    id: 4,
    text: 'Starred',
    color: 'ffdf32'
  },
  draft: {
    id: 5,
    text: 'Draft',
    color: '666666'
  },
  trash: {
    id: 6,
    text: 'Trash',
    color: 'b00e0e'
  },
  search: {
    id: -1,
    text: 'Search',
    color: '000000'
  },
  allmail: {
    id: -1,
    text: 'All Mail',
    color: null
  }
};

export const requiredMinLength = {
  username: 3,
  fullname: 1,
  password: 8
};

export const requiredMaxLength = {
  username: 255,
  fullname: 255,
  password: 255
};
