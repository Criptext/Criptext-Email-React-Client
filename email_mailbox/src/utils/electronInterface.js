import { labels } from './systemLabels';
import { cleanDataLogout as cleanData } from './ipc';

const { remote } = window.require('electron');
const { getCurrentWindow } = remote;

const newsClient = remote.require('./src/newsClient');
const globalManager = remote.require('./src/globalManager');

export const myAccount = remote.require('./src/Account');
export const mySettings = remote.require('./src/Settings');

export const { requiredMinLength, requiredMaxLength } = remote.require(
  './src/validationConsts'
);

export const { FILE_SERVER_APP_ID, FILE_SERVER_KEY } = remote.require(
  './src/utils/const'
);

const additionalLabels = {
  search: {
    id: -2,
    text: 'Search'
  },
  allmail: {
    id: -1,
    text: 'All Mail'
  }
};
export const LabelType = Object.assign(labels, additionalLabels);

export const setInternetConnectionStatus = status => {
  globalManager.internetConnection.setStatus(status);
};

export const getDeviceType = () => globalManager.deviceType.id;

/*  Window events
----------------------------- */
export const reloadWindow = () => {
  getCurrentWindow().reload();
};

export const checkDisableRequests = () => {
  return globalManager.windowsEvents.checkDisabled();
};

export const disableEventRequests = () => {
  globalManager.windowsEvents.disable();
};

export const enableEventRequests = () => {
  globalManager.windowsEvents.enable();
};

export const getGettingEventsStatus = () => {
  return globalManager.isGettingEvents.get();
};

export const setGettingEventsStatus = value => {
  globalManager.isGettingEvents.set(value);
};

/*  News Client
----------------------------- */
export const getNews = ({ code }) => {
  return newsClient.getNews({ code });
};

/*  DataBase
----------------------------- */
export const cleanDataLogout = async ({ recipientId }) => {
  return await cleanData({ recipientId });
};
