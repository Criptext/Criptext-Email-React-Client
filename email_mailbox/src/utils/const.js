/*global process */
import { LabelType } from './electronInterface';
import { currentLanguage } from './../lang';

export const appDomain =
  process.env.NODE_ENV === 'development'
    ? process.env.REACT_APP_APPDOMAIN
    : 'criptext.com';

export const avatarBaseUrl =
  process.env.NODE_ENV === 'development'
    ? process.env.REACT_APP_AVATAR_URL
    : `https://api.criptext.com/user/avatar/`;

export const avatarBorderUrl =
  process.env.NODE_ENV === 'development'
    ? process.env.REACT_APP_BORDER_URL
    : `https://api.criptext.com/user/avatar/frame/`;

export const adminUrl =
  process.env.NODE_ENV === 'development'
    ? process.env.REACT_APP_ADMIN_URL
    : `https://admin.criptext.com`;

export const unsentText = 'This content was unsent';

export const sidebarItemsOrder = [
  LabelType.inbox.id,
  LabelType.sent.id,
  LabelType.draft.id,
  LabelType.starred.id,
  LabelType.spam.id,
  LabelType.trash.id
];

export const IconLabels = {
  [LabelType.inbox.id]: {
    icon: 'icon-mailbox'
  },
  [LabelType.sent.id]: {
    icon: 'icon-sent'
  },
  [LabelType.draft.id]: {
    icon: 'icon-doc'
  },
  [LabelType.starred.id]: {
    icon: 'icon-star-fill'
  },
  [LabelType.spam.id]: {
    icon: 'icon-not'
  },
  [LabelType.trash.id]: {
    icon: 'icon-trash'
  },
  allmail: {
    icon: 'icon-mail',
    text: LabelType.allmail.text
  }
};

export const SectionType = {
  MAILBOX: 1,
  THREAD: 2,
  SETTINGS: 3
};

export const FeedItemType = {
  OPENED: {
    value: 1,
    icon: 'icon-double-checked'
  },
  DOWNLOADED: {
    value: 2,
    icon: 'icon-attach'
  }
};

export const EmailStatus = {
  FAIL: 1,
  UNSEND: 2,
  NONE: 3,
  SENDING: 4,
  SENT: 5,
  DELIVERED: 6,
  READ: 7
};

export const SocketCommand = {
  NEW_EMAIL: 101,
  EMAIL_TRACKING_UPDATE: 102,
  ATTACHMENT_TRACKING_UPDATE: 103,
  SEND_EMAIL_ERROR: 104,
  NEW_DRAFT: 105,
  EMAIL_MUTED: 106,
  LOW_PREKEYS_AVAILABLE: 107,

  DEVICE_LINK_AUTHORIZATION_REQUEST: 201,
  DEVICE_LINK_AUTHORIZATION_CONFIRMATION: 202,
  KEYBUNDLE_UPLOADED: 203,
  DATA_UPLOADED: 204,
  DEVICE_REMOVED: 205,
  DEVICE_LINK_AUTHORIZATION_DENY: 206,
  DEVICE_LINK_REQUEST_RESPONDED: 207,

  SYNC_DEVICE_REQUEST: 211,
  SYNC_DEVICE_ACCEPT: 212,
  SYNC_DEVICE_REJECT: 216,
  SYNC_DEVICE_REQUEST_RESPONDED: 217,

  PEER_EMAIL_READ_UPDATE: 301,
  PEER_EMAIL_UNSEND: 307,

  PEER_THREAD_READ_UPDATE: 302,
  PEER_EMAIL_LABELS_UPDATE: 303,
  PEER_THREAD_LABELS_UPDATE: 304,
  PEER_EMAIL_DELETED_PERMANENTLY: 305,
  PEER_THREAD_DELETED_PERMANENTLY: 306,
  PEER_LABEL_CREATED: 308,
  PEER_USER_NAME_CHANGED: 309,
  PEER_PASSWORD_CHANGED: 310,
  PEER_RECOVERY_EMAIL_CHANGED: 311,
  PEER_RECOVERY_EMAIL_CONFIRMED: 312,
  PEER_AVATAR_CHANGED: 313,
  PEER_LABEL_UPDATE: 319,
  PEER_LABEL_DELETE: 320,
  PEER_SET_BLOCK_REMOTE_CONTENT: 326,
  PEER_SET_TRUSTED_EMAIL: 327,

  NEW_ANNOUNCEMENT: 401,
  UPDATE_AVAILABLE: 402,
  UPDATE_DEVICE_TYPE: 403,
  ACTION_REQUIRED: 404,

  SEND_OPEN_EVENT: 500,

  SUSPENDED_ACCOUNT_EVENT: 600,
  REACTIVATED_ACCOUNT_EVENT: 601,

  CUSTOMER_TYPE_UPDATE: 700,
  ADDRESS_CREATED: 701,
  ADDRESS_STATUS_UDATE: 702,
  ADDRESS_DELETED: 703,
  CUSTOM_DOMAIN_CREATED: 704,
  CUSTOM_DOMAIN_DELETED: 705,
  DEFAULT_ADDRESS: 706,
  ADDRESS_NAME_UPDATE: 707
};

export const NOTIFICATION_ACTIONS = {
  ANTI_PUSH: 'anti_push',
  NEW_EMAIL: 'open_thread',
  OPEN_EMAIL: 'open_activity'
};

export const deviceTypes = {
  PC: 1,
  IOS: 2,
  ANDROID: 3
};

export const usefulLinks = {
  FAQ: `https://criptext.com/${currentLanguage}/faq/`,
  PRIVACY_POLICY: `https://www.criptext.com/${currentLanguage}/privacy`,
  TERMS_OF_SERVICE: `https://www.criptext.com/${currentLanguage}/terms`,
  CRIPTEXT_LIBRARIES: `https://www.criptext.com/${currentLanguage}/open-source-desktop`
};

export const composerEvents = {
  EDIT_DRAFT: 'edit-draft',
  FORWARD: 'forward',
  REPLY: 'reply',
  REPLY_ALL: 'reply-all',
  NEW_WITH_DATA: 'new-with-data'
};

export const formInviteFriendEmailContent = () => {
  return '<!DOCTYPE html><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Hey, try Criptext now!</title><style type="text/css">.boton_1{text-decoration: none;padding: 8px;padding-left: 15px;padding-right: 15px;font-family: Avenir Next;font-weight: 500;font-size: 15px;color: #ffffff;background-color: #0091ff;border-radius: 26px;}.boton_1:hover{opacity: 0.9;text-decoration: none;}.boton_2{color: #0091ff}#criptext-website{color: #8FD8FF;text-decoration: none;}#confirm-button{color: #ffffff;}a:active, a:visited{color: #ffffff;}</style></head><body><br/><br/><div style="background-color: #F0F0F0; padding: 6%; max-width: 100%"><div style="background-color:#ffffff; padding-top:60px; padding-bottom: 60px; height:auto; max-width:100%; text-align: center; margin: 0 auto; font-family: avenir next;" ><div style="width: 80%; margin: 0 auto;"><div style="margin: 0 auto;"><img src="https://s3-us-west-2.amazonaws.com/web-res/Emails/header-icon.png" height="49px" ></div><div style="margin: 0 auto; padding-top:25px "><h2 style="color:#373a45; font-weight: 600; font-size: 23px; text-align: center">Hey, try Criptext now!</h2></div><div style="width: 100%; padding-top:20px "><p style="font-size: 15px; color:#9b9b9b; text-align: center">Criptext is an encrypted email service that guarantees security, privacy and control over all your email communications. We don‘t have access to your emails nor do we store them in our servers. <br/>You‘re in control now.</p></div><div style="margin: 0 auto; padding-top:30px" ><a id="confirm-button" href="http://www.criptext.com/dl" class="boton_1">Download</a></div></div></div><div style="max-width: 100%; margin: 5px auto; background-image: url(https://s3-us-west-2.amazonaws.com/web-res/Emails/footer-background.png); height: 110px; background-position: center top; background-repeat: no-repeat; background-size: auto 200%; font-family: avenir next; text-align: center;"><table style="width: 65%; margin: 0 auto; text-align: left; color: #ffffff; font-size: 13px; padding-top: 35px;"><tr><td style="font-weight: 600;">Encrypted. Private. Simple.<br><a href="https://criptext.com/" target="_blank" id="criptext-website" style="color: #8FD8FF; font-weight: 400;">www.criptext.com</a></td><td><a href="https://twitter.com/Criptext" target="_blank"><img src="https://s3-us-west-2.amazonaws.com/web-res/Emails/twitter.png" style="height: 18px;"></a></td><td><a href="https://medium.com/criptext" target="_blank"><img src="https://s3-us-west-2.amazonaws.com/web-res/Emails/medium.png" style="height: 18px;"></a></td><td><a href="https://www.instagram.com/criptext/" target="_blank"><img src="https://s3-us-west-2.amazonaws.com/web-res/Emails/instagram.png" style="height: 18px;"></a></td></tr></table></div></div></body></html>';
};

export const EXTERNAL_RECIPIENT_ID_SERVER = 'bob';

export const SEND_BUTTON_STATUS = {
  DISABLED: 1,
  ENABLED: 2
};
