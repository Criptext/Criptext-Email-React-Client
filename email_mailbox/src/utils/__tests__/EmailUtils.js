/* eslint-env node, jest */

import {
  addCollapseDiv,
  checkEmailIsTo,
  compareEmailDate,
  defineRejectedLabels,
  filterCriptextRecipients,
  getRecipientIdFromEmailAddressTag,
  getRecipientsFromData,
  cleanEmailBody
} from '../EmailUtils';
import { LabelType, myAccount } from '../electronInterface';
import { appDomain } from './../const';

jest.mock('./../../utils/const');
jest.mock('./../../utils/electronInterface');
jest.mock('./../../utils/ipc');

describe('Define rejected labels ', () => {
  it('Define rejected labels for Spam label', () => {
    const labelId = LabelType.spam.id;
    const expectedResult = [LabelType.trash.id];
    const rejectedlabels = defineRejectedLabels(labelId);
    expect(rejectedlabels).toEqual(expectedResult);
  });

  it('Define rejected labels for Trash label', () => {
    const labelId = LabelType.trash.id;
    const expectedResult = [LabelType.spam.id];
    const rejectedlabels = defineRejectedLabels(labelId);
    expect(rejectedlabels).toEqual(expectedResult);
  });

  it('Define rejected labels for All mail label', () => {
    const labelId = LabelType.allmail.id;
    const expectedResult = [
      LabelType.spam.id,
      LabelType.trash.id,
      LabelType.draft.id
    ].sort();
    const rejectedlabels = defineRejectedLabels(labelId).sort();
    expect(rejectedlabels).toEqual(expectedResult);
  });

  it('Define rejected labels for another label', () => {
    const expectedRejectedlabels = [LabelType.spam.id, LabelType.trash.id];

    const inboxLabelId = LabelType.inbox.id;
    const inboxRejectedlabels = defineRejectedLabels(inboxLabelId).sort();

    const sentLabelId = LabelType.sent.id;
    const sentRejectedlabels = defineRejectedLabels(sentLabelId).sort();

    const draftLabelId = LabelType.draft.id;
    const draftRejectedlabels = defineRejectedLabels(draftLabelId).sort();

    const starredLabelId = LabelType.starred.id;
    const starredRejectedlabels = defineRejectedLabels(starredLabelId).sort();

    expect(inboxRejectedlabels).toEqual(expectedRejectedlabels);
    expect(sentRejectedlabels).toEqual(expectedRejectedlabels);
    expect(draftRejectedlabels).toEqual(expectedRejectedlabels);
    expect(starredRejectedlabels).toEqual(expectedRejectedlabels);
  });
});

describe('Sort emails ', () => {
  it('Should sort emails by date', () => {
    const emails = [
      { date: '2018-10-04 21:46:55' },
      { date: '2018-10-04 21:36:55' },
      { date: '2018-10-04 21:26:55' }
    ];

    const result = emails.sort(compareEmailDate);
    expect(result).toEqual([
      { date: '2018-10-04 21:26:55' },
      { date: '2018-10-04 21:36:55' },
      { date: '2018-10-04 21:46:55' }
    ]);
  });
});

describe('Add collapse div ', () => {
  it('Should add div collpase to html with: <blockquote>', () => {
    const html =
      '<div><blockquote><p></p><br/><blockquote></blockquote></blockquote></div>';
    const result = addCollapseDiv(html, 1);
    expect(result).toMatchSnapshot();
  });

  it('Should add div collpase to html with: <div class="criptext_quote">', () => {
    const html =
      '<div class="criptext_quote"><blockquote><p></p><br/><blockquote></blockquote></blockquote></div>';
    const result = addCollapseDiv(html, 1);
    expect(result).toMatchSnapshot();
  });

  it('Should add div collpase to html with: <div class="gmail_quote">', () => {
    const html =
      '<div class="gmail_quote"><blockquote><p></p><br/><blockquote></blockquote></blockquote></div>';
    const result = addCollapseDiv(html, 1);
    expect(result).toMatchSnapshot();
  });
});

describe('Get recipientId from EmailAddressTag ', () => {
  it('Should get recipientId from criptext domain', () => {
    const from = `"Alice$$" <Alice@${appDomain}>`;
    const recipientId = getRecipientIdFromEmailAddressTag(from);
    const result = 'alice';
    expect(recipientId).toEqual(result);
  });

  it('Should get recipientId from external domain', () => {
    const from = 'Bob 8989 <Bob@domain.com>';
    const recipientId = getRecipientIdFromEmailAddressTag(from);
    const result = 'bob@domain.com';
    expect(recipientId).toEqual(result);
  });

  it('Should get recipientId from external domain, with name comma added', () => {
    const from = `Lola, <lola@domain.com>`;
    const recipientId = getRecipientIdFromEmailAddressTag(from);
    const result = 'lola@domain.com';
    expect(recipientId).toEqual(result);
  });

  it('Should get recipientId from external domain, with name as emailadresstag', () => {
    const from = `<alice@${appDomain}> <lola@domain.com>`;
    const recipientId = getRecipientIdFromEmailAddressTag(from);
    const result = 'lola@domain.com';
    expect(recipientId).toEqual(result);
  });

  it('Should get recipientId from external domain, with emailadresstag without tag', () => {
    const from = `alice@domain.com`;
    const recipientId = getRecipientIdFromEmailAddressTag(from);
    const result = 'alice@domain.com';
    expect(recipientId).toEqual(result);
  });
});

describe('Filter emails by domain ', () => {
  it('Should filter emails by a different domain defined', () => {
    const recipients = [
      `userA@${appDomain}`,
      `userB@${appDomain}`,
      'userC@xxx.com'
    ];
    const filtered = filterCriptextRecipients(recipients);
    const result = [`userA@${appDomain}`, `userB@${appDomain}`];
    expect(filtered).toEqual(result);
  });
});

describe('Sanitize emails ', () => {
  it('Should clear unauthorized tags', () => {
    const unsanitizedHtml = `<div><script>alert("I am a malicious script")</script><p>And I'm just a paragraph</p></div>`;
    const expectedHtml = `<div><p>And I'm just a paragraph</p></div>`;
    const sanitizedHtml = cleanEmailBody(unsanitizedHtml);
    expect(sanitizedHtml).toEqual(expectedHtml);
  });

  it('Should clear unauthorized attributes', () => {
    const unsanitizedTag = `<a href="http://www.criptext.com" onclick="execMaliciousScript()"> Link </a>`;
    const expectedTag = `<a href="http://www.criptext.com"> Link </a>`;
    const sanitizedTag = cleanEmailBody(unsanitizedTag);
    expect(sanitizedTag).toEqual(expectedTag);
  });

  it('Should not clear authorized schemes on attributes', () => {
    const unsanitizedTag = `<img src="http://www.domain.com/path/to/image.png" />`;
    const expectedTag = `<img src="http://www.domain.com/path/to/image.png" />`;
    const sanitizedTag = cleanEmailBody(unsanitizedTag);
    expect(sanitizedTag).toEqual(expectedTag);
  });

  it('Should clear unauthorized schemes on attributes', () => {
    const unsanitizedTag = `<img src="unknownScheme:data/format" alt="filename.jpg" />`;
    const expectedTag = `<img alt="filename.jpg" />`;
    const sanitizedTag = cleanEmailBody(unsanitizedTag);
    expect(sanitizedTag).toEqual(expectedTag);
  });
});

describe('Form recipients', () => {
  it('Should form recipients', () => {
    const data = {
      to: ['<test@criptext.com>'],
      cc: [
        'Name<>1 <test1@criptext.com>',
        '<error@criptext.com> <test2@criptext.com>',
        'error2@criptext.com <test3@criptext.com>'
      ],
      bcc: ['test4@criptext.com'],
      from: 'Name Surname <test5@criptext.com>'
    };
    const recipients = getRecipientsFromData(data);
    expect(recipients).toMatchSnapshot();

    const oldData = {
      to: '<test@criptext.com>',
      cc:
        'Name<>1 <test1@criptext.com>, <error@criptext.com> <test2@criptext.com>, error2@criptext.com <test3@criptext.com>',
      bcc: 'test4@criptext.com',
      from: 'Name Surname <test5@criptext.com>'
    };
    const oldRecipients = getRecipientsFromData(oldData);
    expect(oldRecipients).toMatchSnapshot();
  });
});

describe('Check email is: to me or from me', () => {
  it('Should valid email as to me', () => {
    const data = {
      to: ['<test@criptext.com>'],
      cc: [
        'Name<>1 <test1@criptext.com>',
        '<error@criptext.com> <test2@criptext.com>',
        'error2@criptext.com <test3@criptext.com>'
      ],
      bcc: ['test4@criptext.com'],
      from: 'Name Surname <test5@criptext.com>'
    };
    const recipients = getRecipientsFromData(data);
    const isToMe = checkEmailIsTo(recipients, myAccount.recipientId);
    expect(isToMe).toBeTruthy();
  });

  it('Should not valid email as to me', () => {
    const data = {
      to: ['<1test@criptext.com>'],
      cc: [
        'Name<>1 <test1@criptext.com>',
        '<error@criptext.com> <test2@criptext.com>',
        'error2@criptext.com <test3@criptext.com>'
      ],
      bcc: ['test4@criptext.com'],
      from: 'Name Surname <test5@criptext.com>'
    };
    const recipients = getRecipientsFromData(data);
    const isToMe = checkEmailIsTo(recipients, myAccount.recipientId);
    expect(isToMe).toBeFalsy();
  });
});
