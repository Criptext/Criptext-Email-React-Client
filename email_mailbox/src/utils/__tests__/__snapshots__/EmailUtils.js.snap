// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add collapse div  Should add div collpase to html with: <blockquote> 1`] = `"<head></head><body><div><div class=\\"cptx-div-collapse cptx-div-expanded\\" id=\\"cptx-div-collapse-1\\"><i class=\\"icon-dots\\"></i></div><blockquote style=\\"display: block;\\" id=\\"blockquote-1\\"><p></p><br><blockquote></blockquote></blockquote></div></body>"`;

exports[`Add collapse div  Should add div collpase to html with: <div class="criptext_quote"> 1`] = `"<head></head><body><div class=\\"cptx-div-collapse cptx-div-expanded\\" id=\\"cptx-div-collapse-1\\"><i class=\\"icon-dots\\"></i></div><div class=\\"criptext_quote\\" style=\\"display: block;\\" id=\\"blockquote-1\\"><blockquote><p></p><br><blockquote></blockquote></blockquote></div></body>"`;

exports[`Add collapse div  Should add div collpase to html with: <div class="gmail_quote"> 1`] = `"<head></head><body><div class=\\"gmail_quote\\"><div class=\\"cptx-div-collapse cptx-div-expanded\\" id=\\"cptx-div-collapse-1\\"><i class=\\"icon-dots\\"></i></div><blockquote style=\\"display: block;\\" id=\\"blockquote-1\\"><p></p><br><blockquote></blockquote></blockquote></div></body>"`;

exports[`Form recipients Should form recipients 1`] = `
Object {
  "bcc": Array [
    "test4@criptext.com",
  ],
  "cc": Array [
    "Name<>1 <test1@criptext.com>",
    "<error@criptext.com> <test2@criptext.com>",
    "error2@criptext.com <test3@criptext.com>",
  ],
  "from": Array [
    "Name Surname <test5@criptext.com>",
  ],
  "to": Array [
    "test@criptext.com",
  ],
}
`;

exports[`Form recipients Should form recipients 2`] = `
Object {
  "bcc": Array [
    "test4@criptext.com",
  ],
  "cc": Array [
    "Name<>1 <test1@criptext.com>",
    "<error@criptext.com> <test2@criptext.com>",
    "error2@criptext.com <test3@criptext.com>",
  ],
  "from": Array [
    "Name Surname <test5@criptext.com>",
  ],
  "to": Array [
    "<test@criptext.com>",
  ],
}
`;
